src/webapp/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback } from "react";
+import { Checkbox as MUICheckbox, InputLabel, FormHelperText } from "@material-ui/core";
+import styled from "styled-components";
+
+type CheckboxProps = {
+    id: string;
+    label?: string;
+    checked: boolean;
+    onChange: (isChecked: boolean) => void;
+    helperText?: string;
+    disabled?: boolean;
+    indeterminate?: boolean;
+    errorText?: string;
+    error?: boolean;
+    required?: boolean;
+};
+
+export const Checkbox: React.FC<CheckboxProps> = React.memo(
+    ({
+        id,
+        label,
+        checked,
+        onChange,
+        helperText = "",
+        disabled = false,
+        indeterminate = false,
+    }) => {
+        const handleChange = useCallback(
+            (event: React.ChangeEvent<HTMLInputElement>) => {
+                onChange(event.target.checked);
+            },
+            [onChange]
+        );
+
+        return (
+            <Container>
+                <CheckboxWrapper>
+                    <MUICheckbox
+                        id={id}
+                        checked={checked}
+                        indeterminate={indeterminate}
+                        onChange={handleChange}
+                        disabled={disabled}
+                        size="small"
+                        inputProps={{ "aria-label": label || `${id}-label` }}
```

Comment:
[do not change] [best practices] Just as note, as the object is created on each render, inputProps will be also different and it will issue a re-render. Here it's a small component and it's ok, but for big components it's important.